recipes/cairo/meson/conanfile.py

```diff
@@ -224,166 +230,137 @@ def package(self):
             meson = self._configure_meson()
             meson.install()
         self._fix_library_names()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.pdb")
+        files.rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        files.rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
 
     def package_info(self):
-        self.cpp_info.components["cairo_"].names["pkg_config"] = "cairo"
+        base_requirements = {"pixman::pixman"}
+        base_system_libs = {}
+
+        def add_component_and_base_requirements(component, requirements, system_libs=None):
+            self.cpp_info.components[component].set_property("pkg_config_name", component)
+            self.cpp_info.components[component].names["pkg_config"] = component
+            self.cpp_info.components[component].requires += ["cairo_"] + requirements
+            base_requirements.update(set(requirements))
+            if system_libs is not None:
+                self.cpp_info.components[component].system_libs += system_libs
+                base_system_libs.update(set(system_libs))
+
+        self.cpp_info.set_property("pkg_config_name", "cairo-all-do-no-use")
+        self.cpp_info.names["pkg_config"] = "cairo-all-do-no-use"
         self.cpp_info.components["cairo_"].libs = ["cairo"]
```

Comment:
Don't forget to add a pkg-config name to `cairo_`
```python
self.cpp_info.components["cairo_"].set_property("pkg_config_name", "cairo")
```
Many libraries search for the `cairo` pkgconfig file.